pkg/cloud/scope/session.go

```diff
@@ -20,18 +20,40 @@ import (
 	"sync"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
 	"github.com/aws/aws-sdk-go/aws/session"
 )
 
+// ServiceEndpoint defines a tuple containing AWS Service resolution information
+type ServiceEndpoint struct {
+	ServiceID     string
+	URL           string
+	SigningRegion string
+}
+
 var sessionCache sync.Map
 
-func sessionForRegion(region string) (*session.Session, error) {
+func sessionForRegion(region string, endpoint []ServiceEndpoint) (*session.Session, error) {
 	s, ok := sessionCache.Load(region)
 	if ok {
 		return s.(*session.Session), nil
 	}
 
-	ns, err := session.NewSession(aws.NewConfig().WithRegion(region))
+	resolver := func(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
+		for _, s := range endpoint {
+			if service == s.ServiceID {
+				return endpoints.ResolvedEndpoint{
+					URL:           s.URL,
+					SigningRegion: s.SigningRegion,
+				}, nil
+			}
+		}
+		return endpoints.DefaultResolver().EndpointFor(service, region, optFns...)
+	}
```

Comment:
Move this anonymous function out. Might be the former ruby dev in me, but easier to unit test this way.